src/views/AddInstallation.vue

```diff
@@ -1,30 +1,593 @@
+
+// ---- Installation End Time ----
+const toDateTime = ref(undefined);
+
+const toDT = computed(() => {
+  if (toDateTime.value == null) {
+    console.log("No end date and time defined; returning max. POSIX time.");
+    return dayjs.unix(maxUnixEpoch);
+  } else {
+    return dayjs(toDateTime.value);
+  }
+});
+
+const endIsBeforeNextStart = computed(() => {
+  // The selected instalation end must be earlier than the next installation start.
+  if (fromDateTime.value == null) {
+    return false;
+  } else if (hasInstallations.value) {
+    const fdt_s = fromDT.value.unix();
+    console.log(`from timestamp: ${fdt_s}`);
+    const nextInstallations = installations.value.filter(
+      (inst) => inst.from_timestamp_s > fdt_s
+    );
+    console.log(
+      `next installations: ${nextInstallations} (${typeof nextInstallations})`
+    );
+    if (nextInstallations.length > 0) {
+      const tdt_s = toDT.value.unix();
+      return nextInstallations
+        .map((inst) => inst.from_timestamp_s > tdt_s)
+        .some((elem) => elem);
+    } else {
+      console.log("no next installations.");
+      return true;
+    }
+  } else {
+    console.log("no installations.");
+    return true;
+  }
+});
+
+const endIsValid = computed(() => {
+  const validValue = toDT.value?.isValid();
+  const isLater = toDT.value?.unix() > fromDT.value?.unix();
+  return validValue && isLater && endIsBeforeNextStart.value;
+});
+
+const isInstallationActive = (installation) => {
+  if (installation == null) {
+    return false;
+  } else {
+    let now_s = dayjs().unix();
+    return (
+      installation.from_timestamp_s < now_s &&
+      installation.to_timestamp_s > now_s
+    );
+  }
+};
+
+// ---- Create Installation ----
+const installationDescription = ref(undefined);
+const isPublic = ref(false);
+
+const createInstallation = async () => {
+  let validationErrors = [];
+  // An installation is valid if:
+  // 1. It pertains to a sensor that belongs to the selected organization.
+  if (!isSensorSelected.value) {
+    validationErrors.push(t("installation.errorNoSensor"));
+  }
+  // 2. It has a start time.
+  if (fromDateTime.value == null) {
+    validationErrors.push(t("installation.errorNoStart"));
+  }
+  // 3. The start time lies outsid any existing installation period of the selected sensor
+  if (!startIsOutsideInstalledRanges.value) {
+    validationErrors.push(t("installation.errorOverlappingStart"));
+  }
+  // 4. The end time must be later than the start time.
+  if (!endIsValid.value) {
+    validationErrors.push(t("installation.errorInvalidEnd"));
+  }
+  // 6. If there exist installations with start time later than the current start time, the current stop time must be earlier than the earliest of these start times.
+  if (!endIsBeforeNextStart.value) {
+    validationErrors.push(t("installation.errorOverlappingEnd"));
+  }
+  if (validationErrors.length > 0) {
+    console.log(validationErrors);
+    toast.error(validationErrors.join("\n"));
```

Comment:
We shouldn't use the toast for input validation errors, since they are only shown for a short period of time.
An error for an input field should be shown close to that field as long as the error persists.

Reply:
The toast are shown only when the user clicks the "add installation" button. Actual input validation is handled mainly by the date picker components.